### PR TITLE
Functional safety modifications added

### PR DESCRIPTION
**Pull request by mgong98**
Requesting review and merge of functional safety changes to driver

### CODE REVIEW

**Comment by smallika** [comment]
Reviewed, looks good.

File: drivers/input/touchscreen/atmel_mxt_ts.c [hunk 1]
```diff
@@ -962,11 +1103,125 @@ static int __mxt_write_reg(struct i2c_client *client, u16 reg, u16 len,
 	return ret;
 }
 
+static int __mxt_write_reg_crc(struct i2c_client *client, u16 reg, u16 length,
+			   const void *val, struct mxt_data *data)
+{
+	u8 msgbuf[15];
```
Comment (drxaero) [inline drivers/input/touchscreen/atmel_mxt_ts.c:1109]: Please leave comment why it's 15 bytes. This is the limitation of maXTouch firmware. The 15 bytes are the max number of bytes allowed in a write transfer, including register address (2 bytes), seq. number (1 byte) and CRC8 (1 byte).

**Review by drxaero — COMMENTED** [review]
Hi @mgong98,

Here're some of my comments, FYI.

**Comment by mgong98** [comment]
Yes, T25_address should be u16. This will be changed.

Boolean reset_counter is used to reset the tx_seq_num on a forced reset.

Also I do not under the last comment.

Do you want me to place a comment indicating this is a limit in firmware?

Thanks

File: drivers/input/touchscreen/atmel_mxt_ts.c (hunk 1 above)
Comment (drxaero) [reply drivers/input/touchscreen/atmel_mxt_ts.c:1109]: For me, I have no concern, after all, as a driver developer, we're crafting an adapter between the firmware and the OS. One day if any new comer comes and reads this code, they may ask: "Why it's 15 here?" It's perfectly ok for me to state the "characteristic/limitation" of the firmware in the code to make the driver developers fully understand the code. The better they understand the firmware, the better code they can craft.

If you have concern, please ignore my comment. You lead the way.

**Pull request merged**